packages/flutter/test/material/page_transitions_theme_test.dart

```diff
@@ -97,6 +97,217 @@ void main() {
+    expect(coloredBox.color, Colors.transparent); // Color is transparent during animation.
+  }, variant: TargetPlatformVariant.only(TargetPlatform.android));
+
+  testWidgets('Animation duration changes accordingly when page transition builder changes', (WidgetTester tester) async {
+    Widget buildApp(PageTransitionsBuilder pageTransitionBuilder) {
+      return MaterialApp(
+        theme: ThemeData(
+          pageTransitionsTheme: PageTransitionsTheme(
+            builders: <TargetPlatform, PageTransitionsBuilder>{
+              TargetPlatform.android: pageTransitionBuilder,
+            }
+          )
+        ),
+        routes: <String, WidgetBuilder>{
+          '/': (BuildContext context) => Material(
+            child: TextButton(
+              child: const Text('push'),
+              onPressed: () { Navigator.of(context).pushNamed('/b'); },
+            ),
+          ),
+          '/b': (BuildContext context) => Material(
+            child: Column(
+              mainAxisAlignment: MainAxisAlignment.center,
+              children: <Widget>[
+                TextButton(
+                  child: const Text('pop'),
+                  onPressed: () {
+                    Navigator.of(context).pop();
+                  },
+                ),
+                const Text('page b'),
+              ],
+            )
+          ),
+        },
+      );
+    }
+
+    await tester.pumpWidget(buildApp(const FadeForwardsPageTransitionsBuilder()));
+
+    Finder findFadeForwardsPageTransition() {
+      return find.descendant(
+        of: find.byType(MaterialApp),
+        matching: find.byWidgetPredicate((Widget w) => '${w.runtimeType}' == '_FadeForwardsPageTransition'),
+      );
+    }
+    expect(findFadeForwardsPageTransition(), findsOneWidget);
+
+    await tester.tap(find.text('push'));
+    await tester.pump(const Duration(milliseconds: 799));
+    expect(find.text('page b'), findsNothing);
+    ColoredBox coloredBox = tester.widget(find.byType(ColoredBox).last);
+    expect(coloredBox.color, isNot(Colors.transparent)); // The color is not transparent during animation.
+
+    await tester.pump(const Duration(milliseconds: 801));
+    expect(find.text('page b'), findsOneWidget);
+    coloredBox = tester.widget(find.byType(ColoredBox).last);
+    expect(coloredBox.color, Colors.transparent); // The color is transparent during animation.
+
+    await tester.pumpWidget(buildApp(const FadeUpwardsPageTransitionsBuilder()));
+    await tester.pumpAndSettle();
+    expect(find.descendant(
+      of: find.byType(MaterialApp),
+      matching: find.byWidgetPredicate((Widget w) => '${w.runtimeType}' == '_FadeUpwardsPageTransition'),
+    ), findsOneWidget);
+    await tester.tap(find.text('pop'));
+    await tester.pump(const Duration(milliseconds: 299));
+    expect(find.text('page b'), findsOneWidget);
+    expect(find.byType(ColoredBox), findsNothing); // ColoredBox doesn't exist in CupertinoPageTransition.
```

Comment:
Nit: Should this comment say FadeUpwardsPageTransition instead?

Reply:
Ah thanks for catching this! Updated:)